lib/exercism/user.rb

```diff
@@ -52,21 +52,21 @@ def access?(problem)
   end
 
   # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
-  def self.from_github(id, username, email, avatar_url, joined_as=nil)
-    user = User.where(github_id: id).first
+  def self.from_authentication_data(github_user, joined_as=nil)
```

Comment:
Having now had a look at #3443 I'm a little less confused, but not sure that the things that you've pulled out into this PR are the right things. I think I'll need to have a closer look at what you've done there.